quickwit-metastore/src/metastore/file_backed_metastore/lazy_file_backed_index.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright (C) 2021 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+//! [`FileBackedIndex`] module. It is public so that the crate `quickwit-backward-compat` can
+//! import [`FiledBackedIndex`] and run backward-compatibility tests. You should not have to import
+//! anything from here directly.
+
+use std::sync::{Arc, Weak};
+use std::time::Duration;
+
+use quickwit_storage::Storage;
+use tokio::sync::{Mutex, OnceCell};
+use tracing::error;
+
+use super::file_backed_index::FileBackedIndex;
+use super::store_operations::fetch_index;
+use crate::MetastoreResult;
+
+/// Lazy `FileBackedIndex`. It loads a `FileBackedIndex`
+/// on demand and optionaly spawns a task to poll
+/// regularly the storage and update the index.
+pub struct LazyFileBackedIndex {
+    index_id: String,
+    storage: Arc<dyn Storage>,
+    polling_interval_opt: Option<Duration>,
+    index: OnceCell<Arc<Mutex<FileBackedIndex>>>,
```

Comment:
nitpick: I would rename `index->lazy_index` to highlight the purpose of the `OnceCell<...>`